docs/proposals/11-cluster-health.md

```diff
@@ -0,0 +1,298 @@
+
+##### Possible Actions
+
+- Some resource creations (say, `pods` or `LoadBalancer` `services`) could be throttled/rate-limited.
+
+#### ScaleThresholdReached
+
+If the cluster grows beyond a threshold (a threshold beyond which the cluster health may not be reliably guaranteed), this information can be updated in the conditions with `ScaleThresholdReached` condition type.
+If more information is available about which aspect of the cluster cannot reliably scale anymore (e.g. the control-plane, nodes, loadbalancers etc.) then more specific condition types can be defined.
+
+##### Possible Actions
+
+- Some resource creations (say, `pods` or `LoadBalancer` `services`) could be throttled/rate-limited.
+- Even the request rate to the control-plane could be throttled/rate-limited in an intelligent way to so that critical control-plane and system components get the throughput they need but other non-critical components get throttled/rate-limited.
+- Gardener can defer any disruptive updates until later or ramp them down.
+
+## Alternatives
+
+### Gardener Shoot resource
+
+The `status.conditions` part of the Gardener `Shoot` [resource](#gardener-shoot-resource) could be used to publish and take action on the cluster health conditions.
+
+One possible disadvantage to this approach is that the the `Shoot` resource is hosted in the garden cluster, so any extension or controller that wants to take corrective or mitigating action based the cluster health conditions will need access to the garden cluster. Only, the `gardenlet` needs access to the garden cluster and it might be an anti-pattern to proliferate such access to other controllers in the seed cluster.
+
+A second disadvantage is that the `Shoot` resource is specific to the Gardener and some generic extensions or controllers might prefer to avoid adding a direct dependency to the `Gardener` project.
+
+### Gardener Extensions Cluster Resource
+
+One instance of the Gardener extensions `Cluster` [resource](#gardener-extensions-cluster-resource) is maintained per shoot cluster in the seed cluster whose control-plane is hosted in that seed cluster. So, in principle, this can be used for the purpose of publishing and consuming cluster health conditions.
+
+One of the disadvantages of this approach is that `Cluster` resource is [non-namespaced](https://github.com/gardener/gardener/blob/master/pkg/apis/extensions/v1alpha1/types_cluster.go#L26) which might be suitable for centrally deployed extensions and controllers deployed in the `garden` namespaces but may not be suitable for the shoot-specific controllers deployed per shoot control-plane (deployed in the shoot namespace in the seed cluster).
+
+A second disadvantage is that the part where the shoot status conditions can be maintained is [unstructured](https://github.com/gardener/gardener/blob/master/pkg/apis/extensions/v1alpha1/types_cluster.go#L57) making it unsuitable as an API for publishing and consuming actionable cluster health conditions.
```

Comment:
Hm, I understood that you propose to introduce the **structured** `.status.conditions[]` field (like in other resources we know). Both publishing and consuming conditions is easily possible.